### ISSUE TITLE
Checkbox/List toggle options

### ISSUE THREAD

**Issue by richlewis42** [SBoudrias/Inquirer.js#251]
I have a very long list of options that I use in a check box - they are grouped into categories, and ideally I would like to have only the categories as options, with the option to expand them (e.g. using the right arrow key) to access the embedded options.  Would this be possible to implement, e.g. by supplying choices as an object with keys as categories and associated values as an array of the actual options.

For a silly example:

```javascript
{
    type: 'checkbox',
    message: 'choose something from long list of options',
    choices: {
                      fun_options: ['balloons', 'megaphone', 'laser'],
                      boring_options: ['pencil', 'carpet cleaner', 'dust'],
                      edible_options: ['cake', 'toast', 'bacon']
                  }
}
```

Id rather the prompt didn't start out with 9 options - if you know you won't be choosing a boring option, those are just clutter to scroll past.

Any opinions?

**Comment by SBoudrias** [comment]
Just use two prompts. Depending on the result of the first prompt, then display the second one.

**Comment by richlewis42** [comment]
Thanks for the rapid reply.  I already tried many separate prompts, but there are about 30 different subsections (with ~10 options each), and the typical user would be potentially interested in over 10 of the subsections - it just felt very clunky to have 10 follow up checkbox prompts coming one after another, and an accordion style interface would have felt intuitive when it was implemented that way.

I understand it is an edge use case, just thought I would suggest it as it seems to fit nicely with the current api - I suppose it would be difficult to implement though.
